app/models/category.rb

```diff
@@ -0,0 +1,6 @@
+class Category < ApplicationRecord
+    include Visible
+
```

Comment:
Add enum for statuses. This will automatically validates inclusion  into the list and methods active? and archived? will be generated automatically. No needed to create a separate module.

Reply:
https://stackoverflow.com/questions/32938729/how-to-store-enum-as-string-to-database-in-rails